core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/AutoscaleV4Controller.java

```diff
@@ -175,7 +178,8 @@ public AutoscaleRecommendationV4Response getRecommendation(String crn) {
     }
 
     @Override
-    @CheckPermissionByAccount(action = AuthorizationResourceAction.DATAHUB_READ)
+    @DisableCheckPermissions
+    @PreAuthorize("hasRole('AUTOSCALE')")
```

Comment:
have you tested this? in the future we will remove this `hasRole('AUTOSCALE')` and switch to internal api and periscope can call it internally. But for now, this will be good I think, just want to make sure that you have tested it and normal user (not periscope, not poweruser) cannot call this method

Reply:
yes all **_hasRole('AUTOSCALE')_** methods are only called with internalcrn of periscope and these api are working.